city_scrapers/spiders/chi_citycouncil.py

```diff
@@ -3,142 +3,116 @@
 All spiders should yield data shaped according to the Open Civic Data
 specification (http://docs.opencivicdata.org/en/latest/data/event.html).
 """
-import scrapy
-import requests
-import json
 import datetime
+import json
+from urllib.parse import urljoin, parse_qs
 
-from pytz import timezone
 import dateutil.parser
+import scrapy
 
 from city_scrapers.spider import Spider
 
 
 class Chi_citycouncilSpider(Spider):
     name = 'chi_citycouncil'
-    long_name = "Chicago City Council"
-    ocd_url = 'https://ocd.datamade.us/'
-    ocd_tp = 'events/?'
-    ocd_d = 'start_date__gt=' + str(datetime.date.today()) + '&'
-    ocd_srt = 'sort=start_date&'
-    ocd_jur = 'jurisdiction=ocd-jurisdiction/'
-    ocd_loc = 'country:us/state:il/place:chicago/government'
-
-    allowed_domains = [ocd_url]
-    start_urls = [ocd_url + ocd_tp + ocd_d + ocd_srt + ocd_jur + ocd_loc]
-
-    def __init__(self, session=requests.Session()):
-        """
-        Initialize a spider with a requests session for use in the 
-        _make_ocd_request function.
-        """
-        self.session = session
+    agency_id = 'City Council'
+    timezone = 'America/Chicago'
+    classification = 'City Council'
+    allowed_domains = ['ocd.datamade.us']
+
+    endpoint = "https://ocd.datamade.us/events/"
+    query = {
+        "start_date__gt": str(datetime.date.today()),
+        "sort": "start_date",
+        "jurisdiction": "ocd-jurisdiction/country:us/state:il/place:chicago/government",
+    }
+    # the response doesn't include the address for city hall
+    address = '121 N LaSalle Dr, Chicago, IL'
+
+    def start_requests(self):
+        yield scrapy.FormRequest(url=self.endpoint, method='GET', formdata=self.query, callback=self.parse)
 
     def parse(self, response):
         """
-        This is not a traditional spider, rather, this is a glorified wrapper
+        This is not a traditional spider, rather, this is a wrapper
         around the Open Civic Data API to which the Chicago City Clerk
         Legistar site info has already been scraped.
         We will attempt to return all events that have been uploaded in the
         future, i.e. past today's date.
         """
-        data = json.loads(response.text)
-
-        for item in data['results']:
-            yield self._parse_item(item)
-
-        # self._parse_next(response) yields more (responses to parse
+        data = json.loads(response.body)
```

Comment:
Sounds good to look at tonight. It works when I run the spider locally too, but I'm surprised that it ever works, since `response.body` is bytes and `json.loads` should take a string or unicode, so I'd expect an error. What you suggest looks right or maybe [`response.text`](https://doc.scrapy.org/en/latest/topics/request-response.html#scrapy.http.TextResponse.text), which just looks like shorthand for the decoding that you suggested above.